RHIDP-12596: Personal AI Notebooks (#2182)
redhat-developer/red-hat-developers-documentation-rhdh: modules/shared/proc-enable-data-persistence-for-developer-lightspeed-notebooks.adoc

:_mod-docs-content-type: PROCEDURE

[id="enable-data-persistence-for-developer-lightspeed-notebooks_{context}"]
= Enable data persistence for Developer Lightspeed Notebooks

[role="_abstract"]

> To persist Notebook sessions, documents, and AI history across service restarts, as an administrator, you must configure the Notebooks storage backends to use persistent volumes. 

Reviewer: Remove fluff

```suggestion
To persist Notebook sessions, documents, and AI history across service restarts, configure the Notebooks storage backends to use persistent volumes. 
```

By default, the service uses ephemeral storage in the `/tmp` directory, which the system clears during a pod restart.

.Prerequisites

> * You have administrator privileges to the {product-very-short} instance.

Reviewer: This is lacking clarity: you need permissions on the config files at OpenShift level, but you do not need any permissions at RHDH level. 

Consider using the same sentence as in https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.9/html/configuring_red_hat_developer_hub/provision-and-use-your-custom-rhdh-configuration_configuring-rhdh#provision-your-custom-rhdh-configuration_provision-and-use-your-custom-rhdh-configuration:

By using the [OpenShift CLI (oc)](https://docs.redhat.com/en/documentation/openshift_container_platform/4.21/html-single/cli_tools/index#cli-about-cli_cli-developer-commands), you have access, with developer permissions, to the OpenShift cluster containing your Developer Hub instance.

> * You have access to the configuration file (`llama-stack-configs/config.yaml`).

Reviewer: Remove this item, this is an implicit consequence of the previous prerequisite.

> * A Persistent Volume Claim (PVC) is provisioned in your cluster and mounted to the container (for example, at `/var/lib/lightspeed-data`).

Reviewer: Mounted to which container?

.Procedure

> . Create a custom ConfigMap to store your configuration changes and reference it in your deployment. You must use a custom ConfigMap to ensure changes persist during Operator reconciliation or Helm upgrades.

Reviewer: ```suggestion
. Create a custom config map to store your configuration changes and reference it in your deployment. You must use a custom config map to ensure changes persist during Operator reconciliation or Helm upgrades.
```

> . Create a custom ConfigMap to store your configuration changes and reference it in your deployment. You must use a custom ConfigMap to ensure changes persist during Operator reconciliation or Helm upgrades.

Reviewer: Consider reordering, for consistency with https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.9/html/configuring_red_hat_developer_hub/provision-and-use-your-custom-rhdh-configuration_configuring-rhdh#provision-your-custom-rhdh-configuration_provision-and-use-your-custom-rhdh-configuration

1. Author your custom `llama-stack-configs/config.yaml` file.
2. Provision your custom `llama-stack-configs/config.yaml` file to your OpenShift Container Platform cluster.

```
$ oc create configmap  llama-stack-configs --from-file=llama-stack-configs/config.yaml --namespace=my-rhdh-project
```

Author: I have moved it into the prereq and added the link!

> . Modify storage paths: In your custom ConfigMap, update the `llama-stack-configs/config.yaml` file to point the `kv_notebooks` storage backend to your persistent mount point:

Reviewer: ```suggestion
. Modify storage paths: In your custom config map, update the `llama-stack-configs/config.yaml` file to point the `kv_notebooks` storage backend to your persistent mount point:
```
